flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/GroupAggProcessFunction.scala

```diff
@@ -118,19 +131,33 @@ class GroupAggProcessFunction(
       }
     }
 
-    state.update(accumulators)
+    if (inputCnt > 0) {
```

Comment:
I don't think we need to remove `override def consumesRetractions = true` from `DataStreamGroupAggregate`. The `GroupAggProcessFunction` still consumes retraction messages. If `generateRetraction` is set to `true`, `GroupAggProcessFunction` either produces two messages (acc / retract) or one message (for first acc or last retract). If it is set to `false`, it always produces one message, either for insert/update (with `change = true`) or for delete (with `change = false`).
So `GroupAggProcessFunction` can still work in both modes.

Which tests in `RetractionRulesTest` would you like to update?